### ISSUE TITLE
Failed to load Better BibTex

### ISSUE THREAD

**Issue by ZhengJIN94** [PKM-er/obsidian-zotlit#236]
Hi, 

After I installed ZotLit, there is an error as follow figure

<img width="787" alt="image" src="https://github.com/PKM-er/obsidian-zotlit/assets/38369581/ee676826-1f70-4c0b-b57d-4330764bf741">

I am sure the better-bibtex-search.sqlite is in that directory.

Another issue is that when I use [@ to cite literature, it return back as [@undefined].

It there any way to solve these two problems?

**Comment by retorquere** [comment]
The citation keys have moved. They live in better-bibtex.sqlite now, but that database must be mounted using

```
ATTACH DATABASE '<path to better-bibtex.sqlte>' AS betterbibtex;
```

after which the citation keys can be read from `betterbibtex.citationkey`.

**Comment by aidenlx** [comment]
Noted, will update shortly

**Comment by aidenlx** [comment]
@retorquere could you provide the version of the better bibtex?

**Comment by retorquere** [comment]
6.7.132 or later.

**Comment by benpla** [comment]
I have a similar effect (noted here for clarifying):
It differs in Zotero items created before the update and Zotero items created now.

**OLD Zotero Items:** 
On "Create Literature Note" all is fine: citekey is specified correctly.

**NEW Zotero Items** (added after update):
citekey first looks like it always does in Zotero, but On "Create Literature Note" citekey are "undefined". 

<img width="1934" alt="grafik" src="https://github.com/PKM-er/obsidian-zotlit/assets/38682514/b8ad79c1-6028-4e8e-987f-7c66ab335dd7">


Note: 
If I change an old Citekey and "Create Literature Note", then the change take not into account and the old key is exported.

BibTeX: 6.7.132
Obsidian Note for Zotero: 1.0.1

**Comment by retorquere** [comment]
That's because I have left the old database in place, it's an easy fallback scenario because database changes are the scariest of changes. But that backup is no longer being updated, so zotlit will have to be updated. [This PR](https://github.com/plandes/zotsite/pull/29) fixed it for another project, but I do not understand the composition of zotlit.

**Comment by JacobHamelMottiez** [comment]
I have the same problem as @benpla with citation key of newly added references in Zotero! Thanks @retorquere and @aidenlx for your great work, I am really enjoying both your plugins and I hope this issue will be resolve soon!

**Comment by retorquere** [comment]
I'll be glad to help when @PKM-er works on it, but I don't understand how this plugin hangs together.

**Comment by alephpiece** [comment]
I have to roll back to Better BibTeX 6.7.130 to make this plugin work.

**Comment by retorquere** [comment]
Which will be undone by Zotero in the next 48 hours. This is not a sustainable workaround.

**Comment by retorquere** [comment]
And if you roll back (with all the risk that entails - no support, and when you upgrade back to the regular release, and new citekeys and auto exports you make will be lost during the upgrade), please roll back to 124, not 130.

**Comment by luyanze** [comment]
I use Better BibTeX 6.7.133 and Obsidian Note for Zotero: 1.0.1, and meet the same problems. Roll back is a useful way to temporary solve these problem.

**Comment by alephpiece** [comment]
> Which will be undone by Zotero in the next 48 hours. This is not a sustainable workaround.

I just turned off automatic updates for Better BibTeX. Hope it works.

**Comment by retorquere** [comment]
Still recommend 124 over 130

**Comment by retorquere** [comment]
Just as a reminder then: when you go back to the regular release of BBT, some keys might be regenerated, and some auto exports may be lost.

Also please do install the latest BBT release before opening new issues.

**Comment by mariomui** [comment]
> I tried to prepare a PR but I don't understand the source code of this plugin.

Part of this plugin uses ophidian core which connects this.current to the SettingsPluginClass. 

https://github.com/PKM-er/obsidian-zotlit/blob/dcd41794fd96f92c1866e870821aeb128507b339/app/obsidian/src/services/zotero-db/connector/settings.ts#L9

So here's where the zoterodatadir is being set, and once it is set, that data is being automatically cloned by ophidian core into this.current

**Comment by retorquere** [comment]
That's fine, but I'm not diving into the details of obsidian plugin building, so I'm not going to prepare a PR. If someone who knows how this plugin is put together is willing to make the PR, I can say exactly what must be changed in the database access.

**Comment by mariomui** [comment]
> That's fine, but I'm not diving into the details of obsidian plugin building, so I'm not going to prepare a PR. If someone who knows how this plugin is put together is willing to make the PR, I can say exactly what must be changed in the database access.

https://github.com/PKM-er/obsidian-zotlit/blob/dcd41794fd96f92c1866e870821aeb128507b339/app/obsidian/src/settings/base.ts#L58C35-L58C35

i mean is it as simple as just changing this?

**Comment by retorquere** [comment]
I only see stuff there about paths. I don't see where the connection is made, or where the database is queried.

But if you're willing to make the PR, the change is that instead of connecting directly to better-bibtex.sqlite, you must connect to any other database (`:memory:` would work) and then attach the bbt database using

```
ATTACH ? AS betterbibtex
```

where the parameter is the full path to better-bibtex.slqite. After this you can query `betterbibtex.citekey`.

**Comment by retorquere** [comment]
Oh I see you're fighting with your data right now. That takes precedence of course. Did you sync with the zotero servers? That's easier to restore from.

**Comment by retorquere** [comment]
I test zotero integration with selenium to set up the profile, then I drive zotero using a separate plugin called the debug bridge. Does that answer your question?

**Comment by aidenlx** [comment]
should be fixed in v1.1.3

**Comment by yuhanyao** [comment]
restart obsidian can also solve the problem
